IonicPortals/IonicPortals/PortalsPlugin.swift

```diff
@@ -48,32 +49,40 @@ public class PortalsPlugin: CAPPlugin {
     }
     
     // MARK: Static methods for use by Swift app
-    
+
     public static func subscribe(_ topic: String, _ callback: @escaping (_ result: SubscriptionResult) -> ()) -> Int {
-        PortalsPlugin.subscriptionRef += 1
-        if var subscription = PortalsPlugin.subscriptions[topic] {
-            subscription[PortalsPlugin.subscriptionRef] = callback
-            PortalsPlugin.subscriptions[topic] = subscription
-        } else {
-            let subscription = [PortalsPlugin.subscriptionRef : callback]
-            PortalsPlugin.subscriptions[topic] = subscription
+        queue.sync {
```

Comment:
I'm not like, 100% happy with this. Ideally with a concurrent DispatchQueue we'd have writes performed async behind a barrier flag (like in unsubscribe), but we're currently returning the subscription reference which makes that painful to deal with.